File: java/src/main/java/com/sdl/web/pca/client/Request/IGraphQLRequest.java
package com.sdl.web.pca.client.request;

import java.util.Dictionary;

public interface IGraphQLRequest {
    String Query = null;
    String OperationName = null;
    Dictionary<String, Object> Variables = null;

    /* IGraphQLRequest AddVariable(String name, Object value);*/
    /*SerializationBinder Binder;*/
    /* List<JsonConverter> Convertors;*/

    void setQuery(String query);
    String getQuery();

    void setOperationName(String operationName);
    String getOperationName();

    void setVariables(Dictionary<String, Object> variables);
    Dictionary<String, Object> getVariables();
}

File: net/src/Sdl.Tridion.Api.Client/GraphQLClient/Request/IGraphQLRequest.cs
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Sdl.Tridion.Api.Http.Client;
using Sdl.Tridion.Api.Http.Client.Auth;

namespace Sdl.Tridion.Api.GraphQL.Client.Request
{
    /// <summary>
    /// GraphQL Request
    /// </summary>
    public interface IGraphQLRequest
    {
        /// <summary>
        /// GraphQL Query
        /// </summary>
        string Query { get; set; }

        /// <summary>
        /// GraphQL Operation Name
        /// </summary>
        string OperationName { get; set; }

        /// <summary>
        /// GraphQL Variables
        /// </summary>
        IDictionary<string, object> Variables { get; set; }

        /// <summary>
        /// Authentication used for request
        /// </summary>
        [JsonIgnore]
        IAuthentication Authenticaton { get; set; }

        /// <summary>
        /// Request Headers
        /// </summary>
        [JsonIgnore]
        HttpHeaders Headers { get; set; }

        /// <summary>
        /// Add Variable
        /// </summary>
        /// <param name="name">Variable name</param>
        /// <param name="value">Variable value</param>
        /// <returns>The request</returns>
        IGraphQLRequest AddVariable(string name, object value);

        /// <summary>
        /// Serialization binder used when deserializing the request.
        /// </summary>
        [JsonIgnore]
        SerializationBinder Binder { get; set; }

        /// <summary>
        /// Convertor used when deserialzing the request
        /// </summary>
        [JsonIgnore]
        List<JsonConverter> Convertors { get; set; }

        /// <summary>
        /// Serialize the GraphQL Request
        /// </summary>
        /// <returns>Serialized Request</returns>
        string Serialize();
    }
}
